timeeval/adapters/base.py

```diff
@@ -5,9 +5,26 @@
 
 
 class Adapter(ABC):
-
+    """
+    The base class for all adapters. An adapter is a wrapper around an anomaly detection algorithm that allows to
+    execute it in a standardized way with the TimeEval framework. A subclass of Adapter must implement the _call method
+    that executes the algorithm and returns the results. Optionally, it can also implement the get_prepare_fn and
+    get_finalize_fn methods that are called before and after the execution of the algorithm, respectively.
+    """
     @abstractmethod
     def _call(self, dataset: AlgorithmParameter, args: Dict[str, Any]) -> AlgorithmParameter:  # pragma: no cover
+        """
+        This method runs the anomaly detection algorithm and returns the results.
+
+        Parameters
+        ----------
+
+        dataset : AlgorithmParameter
+            The dataset to run the algorithm on.
+
+        args : Dict[str, Any]
+            The arguments to pass to the algorithm.
```

Comment:
The `args` include TimeEval-internal configuration options as well (e.g. `execution_type`). Here would be a good position to briefly describe them as well because a contributor needs to know this to implement a new `Adapter`:

https://github.com/TimeEval/TimeEval/blob/1e7d8efd1b2472e8f92409a62cf310cb75e04d6c/timeeval/algorithm.py#L95-L105

Reply:
I guess, we should replace this dict with a proper data-class at some point. What do you think?